### DISCUSSION TITLE
我的OJ

### DISCUSSION THREAD

**Discussion by langningchen** [XMOJ-Script-dev/XMOJ-Script#344] [Show and tell]
域名：`http://oj.chenlangning.me`
IP：`20.222.111.12`
开放端口：`22` `80`

欢迎来找BUG @boomzero @PythonSmall-Q 

注：
- 现在邮件服务器有点问题，无法发送邮件验证码
- 如果访问不上因为我为了省钱把它关了，联系我让我开启即可

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/2721870e-be75-47b8-81a3-1dcde64351ab)

Running...

**Comment by boomzero** [comment]
~start ddos~

**Comment by langningchen** [comment]
**注意：访问的时候不要开代理，不支持 HTTPS**

**Comment by langningchen** [comment]
> ~start ddos~
@boomzero 试试看，不过真的可能会崩

**Comment by langningchen** [comment]
@boomzero C++写的Socket真的经不起 DDOS

**Comment by boomzero** [comment]
> **注意：访问的时候不要开代理**

我开了, 没事啊

**Comment by langningchen** [comment]
@boomzero 我的代理访问起来会很慢，而且有的时候 502

**Comment by langningchen** [comment]
我先给你个账号

**Comment by boomzero** [comment]
~突然发现你没关ssh password auth,  开始暴力破解密码~

**Comment by langningchen** [comment]


**Reply by boomzero** [reply]
thx

**Comment by langningchen** [comment]
> ~突然发现你没关ssh password auth, 开始暴力破解密码~

因为我不想用私钥，所以我就用密码登陆的

**Comment by boomzero** [comment]
因为有密码一堆人试图破解,所以我一直用私钥

**Comment by langningchen** [comment]
@boomzero 但是我用的是强密码，所以我估计你破解不了

**Comment by langningchen** [comment]
我这里没有显示你登录进去了呢

**Comment by boomzero** [comment]
一直400

**Comment by langningchen** [comment]
我这里也显示出来了

**Comment by langningchen** [comment]
确实，你的HTTP请求是不是有问题

**Reply by boomzero** [reply]
不是啊

**Reply by langningchen** [reply]
@boomzero 我这里很正常

**Comment by langningchen** [comment]
或者你是不是用的HTTPS？

**Comment by langningchen** [comment]
在解析HTTP请求的时候找不到请求Verb

![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/c18b6486-62be-4de5-b636-c94ebb386291)

**Comment by boomzero** [comment]
Safari

**Comment by langningchen** [comment]
你等一分钟，我加一个代码，让他把请求内容打印出来调试一下

**Comment by boomzero** [comment]
<img width="568" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/0e5697c8-bc7a-4a7e-9e2f-35e14ac71447">

**Comment by langningchen** [comment]
@boomzero 加油！

**Comment by boomzero** [comment]
[STATUS] 44.00 tries/min, 44 tries in 00:01h, 474508 to do in 179:45h, 4 active

**Comment by langningchen** [comment]
请求应该成功了吧

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/8efb7334-9f8a-41a7-8f41-36b339b09ced)

**Comment by langningchen** [comment]
全是 curl 
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/359229f5-d366-45e0-9891-c7ae43b25500)

**Reply by boomzero** [reply]
guess why

**Reply by langningchen** [reply]
@boomzero 我不用猜，我本来就知道

**Comment by langningchen** [comment]
@boomzero 你记得换 UA ，我准备屏蔽 curl 的 UA 了

**Comment by langningchen** [comment]
显示你登录进来了

**Reply by boomzero** [reply]
yes

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/2200949c-2317-44e3-a26a-7e57c6214c05)

Wayback Machine 竟然在爬取我的网站！

**Reply by boomzero** [reply]
因为你发在github上了

**Comment by boomzero** [comment]
全是400...

**Comment by langningchen** [comment]
@boomzero 我看到 400 了

**Comment by langningchen** [comment]
因为他收到了空的请求

![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/6677eb81-0323-4586-8eb8-563f8600a51a)

**Comment by langningchen** [comment]
为什么请求是空的呢？肯定因为我的 socket 写的有问题

**Comment by boomzero** [comment]
<img width="1436" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/d41dd0e0-c88c-43a8-8202-0b6c8ae90e62">

**Comment by langningchen** [comment]
@boomzero 我把服务器关了

**Reply by boomzero** [reply]
I know

**Comment by boomzero** [comment]
但是我的submisson一直waiting

**Comment by langningchen** [comment]
@boomzero 80 端口绑定不了了

**Comment by langningchen** [comment]
不知道什么程序占用了

![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/764c1cb8-289b-4fc6-9907-f09fa17e30bb)

我要检查一下

**Comment by langningchen** [comment]
你一会儿注意一下，一个一个提交，我的评测机不太高级，一个没评测完给他第二个就会冲突，两个都是 Waiting 或者 Compiled 的状态

**Comment by langningchen** [comment]
@boomzero 重启了

**Comment by boomzero** [comment]
show and tell: 我正在写的ide:

<img width="1012" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/3eecaf49-27df-4a83-9eb8-44efaf0e6dd8">
(很好看对不对……)

**Reply by langningchen** [reply]
确实

**Reply by langningchen** [reply]
你用的哪里的库？

**Reply by boomzero** [reply]
SwiftUI

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/978ae840-3d2d-48e3-a880-9db2955e097a)

**Comment by langningchen** [comment]
帮你又交了一遍

**Comment by boomzero** [comment]
?

**Comment by boomzero** [comment]
我a+b写挂了

**Comment by langningchen** [comment]
你注意数据范围了吗?

**Comment by langningchen** [comment]
还是我数据范围写错了？

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/22d3f4fb-aec3-4aa3-b8f3-6e2752a58c3a)

**Comment by langningchen** [comment]
服务器还真炸了

![Uploading image.png…]()

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/1420c7cf-a577-4d89-86d7-3532deeb1a85)

**Comment by boomzero** [comment]
<img width="1436" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/7bfdee11-073e-486f-b053-71c6cf23d468">

**Comment by langningchen** [comment]
啊，我怎么把我的Token暴露给你了

**Comment by langningchen** [comment]
@boomzero 我刚才重启了一下，因为改了一下服务器的代码

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/5c388b07-294b-4914-a51a-227d179dfc75)

我把服务器改坏了…… 用GDB调试一下

**Comment by langningchen** [comment]
找到了！ Buffer 太小了

![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/36a30fc9-a95d-4a7f-9b15-8003b57bcee7)

**Comment by langningchen** [comment]
好了

**Comment by langningchen** [comment]
又重启了

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/3ddeff34-a437-4669-a14f-0b3d254876cb)

这是谁的Socket，为什么总是出问题

**Comment by langningchen** [comment]
不过不影响服务器正常运行都不是大问题

**Comment by langningchen** [comment]
183.192.50.5

**Reply by boomzero** [reply]
what?

**Reply by langningchen** [reply]
@boomzero 是你的IP吗？

**Reply by boomzero** [reply]
没事,这个经常换

**Reply by langningchen** [reply]
普陀区

**Reply by boomzero** [reply]
不是

**Reply by langningchen** [reply]
没事，不管他

**Reply by boomzero** [reply]
其实, 闵行区

**Comment by langningchen** [comment]
@boomzero 你再交几道题

**Comment by langningchen** [comment]
@boomzero 这样，我给你一个管理员权限

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/a475e31b-eb2e-48ec-bea6-5ab732d0bef6)

我给了，你重登一下

**Comment by boomzero** [comment]
> 这是谁的Socket，为什么总是出问题


<img width="682" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/1fc80793-24e0-43ce-b591-4a85c16e3a09">

**Comment by langningchen** [comment]
哦，要来是端口扫描啊

**Comment by langningchen** [comment]
给你了Admin别乱改设置就好

**Comment by boomzero** [comment]
你能加https吗

**Comment by langningchen** [comment]
> 你能加https吗

@boomzero 要用 OpenSSL ，难写

**Reply by boomzero** [reply]
不需要,只要装一个server(like nginx) reverse proxy就行了

**Comment by boomzero** [comment]
不过我喜欢用https://caddyserver.com/

**Comment by langningchen** [comment]
反代？还要装nginx？我的磁盘才20G，已经用完94%了

**Comment by langningchen** [comment]
我正在装 kali

**Comment by boomzero** [comment]
caddy可以自动申请https证书

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/f2fdb364-75ab-4b41-a869-16f4d8e40089)

**Comment by langningchen** [comment]
@boomzero 我看到 caddy 的官网简介了

**Comment by langningchen** [comment]
@boomzero 你重新登陆一下，我说了给你分配了管理员权限

**Reply by boomzero** [reply]
我加了一个problem

**Comment by boomzero** [comment]
<img width="1798" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/9ea5716b-da4e-47a7-ba04-61d554a48b8c">

**Comment by langningchen** [comment]
@boomzero 看到了

**Comment by langningchen** [comment]
![image](https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/81672884/f7ada37a-0d1d-40e0-9bc9-3e57476abd8c)

@boomzero

**Comment by boomzero** [comment]
@langningchen 

<img width="1012" alt="image" src="https://github.com/XMOJ-Script-dev/XMOJ-Script/assets/85378277/a12ac823-82de-4e56-8af6-e1fc0951a963">

~多么好看的ide啊~

**Comment by boomzero** [comment]
@langningchen 你能开一下你的OJ吗, 我好像发现了一个安全漏洞

**Comment by langningchen** [comment]
@boomzero 明天吧

**Comment by langningchen** [comment]
今天晚上要睡觉了

**Comment by boomzero** [comment]
Emm.

**Comment by boomzero** [comment]
你这个可以把你整个数据库下载

**Comment by langningchen** [comment]
但是今天下午这一会儿可能不太行，我大概要下午三点以后才能给你把电脑打开了
